homeassistant/components/webostv/__init__.py

```diff
@@ -82,9 +85,48 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+class CannotConnect(Exception):
+    """Error to indicate we cannot connect."""
+
+
 async def async_setup(hass, config):
     """Set up the LG WebOS TV platform."""
-    hass.data[DOMAIN] = {}
+    hass.data.setdefault(DOMAIN, {})
+    if DOMAIN not in config:
+        return True
+
+    for conf in config[DOMAIN]:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set the config entry up."""
+    if not config_entry.options:
+        config = config_entry.data
+        options = {}
+
+        # Get Turn_on service
+        options[CONF_ON_ACTION] = config.get(CONF_ON_ACTION)
+
+        # Get Preferred Sources
+        if sources := config.get(CONF_CUSTOMIZE, {}).get(CONF_SOURCES):
+            options[CONF_SOURCES] = sources
+            if isinstance(sources, list) is False:
+                options[CONF_SOURCES] = sources.split(",")
+
+        hass.config_entries.async_update_entry(config_entry, options=options)
```

Comment:
I'd probably make this its own function `_async_migrate_options` since it doesn't have much to do with setting up the entry.